From e721168ad1eb8870bc741f92e0528390dc43cf25 Mon Sep 17 00:00:00 2001
From: Gerrit Pape <pape@smarden.org>
Date: Mon, 21 Jan 2008 14:41:12 +0000
Subject: [PATCH 2/5] bug#438793, #494505: fallback to 'editor' not 'vi'

In editor.c, git-add--interactive.perl, git-sh-setup.sh, git-svn.perl,
t/t7005-editor.sh, contrib/fast-import/git-p4 fallback to 'editor', not
'vi', if $VISUAL and $EDITOR are unset.
---
 contrib/fast-import/git-p4 |    2 +-
 editor.c                   |    2 +-
 git-add--interactive.perl  |    2 +-
 git-sh-setup.sh            |    4 ++--
 git-svn.perl               |    2 +-
 t/t7005-editor.sh          |   14 +++++++-------
 6 files changed, 13 insertions(+), 13 deletions(-)

diff --git a/contrib/fast-import/git-p4 b/contrib/fast-import/git-p4
index e710219..8a17ef5 100755
--- a/contrib/fast-import/git-p4
+++ b/contrib/fast-import/git-p4
@@ -729,7 +729,7 @@ class P4Submit(Command):
             tmpFile.write(submitTemplate + separatorLine + diff + newdiff)
             tmpFile.close()
             mtime = os.stat(fileName).st_mtime
-            defaultEditor = "vi"
+            defaultEditor = "editor"
             if platform.system() == "Windows":
                 defaultEditor = "notepad"
             if os.environ.has_key("P4EDITOR"):
diff --git a/editor.c b/editor.c
index 4d469d0..4823e74 100644
--- a/editor.c
+++ b/editor.c
@@ -19,7 +19,7 @@ int launch_editor(const char *path, struct strbuf *buffer, const char *const *en
 		return error("Terminal is dumb but no VISUAL nor EDITOR defined.");
 
 	if (!editor)
-		editor = "vi";
+		editor = "editor";
 
 	if (strcmp(editor, ":")) {
 		size_t len = strlen(editor);
diff --git a/git-add--interactive.perl b/git-add--interactive.perl
index 8ce1ec9..0898fe2 100755
--- a/git-add--interactive.perl
+++ b/git-add--interactive.perl
@@ -991,7 +991,7 @@ EOF
 	close $fh;
 
 	my $editor = $ENV{GIT_EDITOR} || $repo->config("core.editor")
-		|| $ENV{VISUAL} || $ENV{EDITOR} || "vi";
+		|| $ENV{VISUAL} || $ENV{EDITOR} || "editor";
 	system('sh', '-c', $editor.' "$@"', $editor, $hunkfile);
 
 	if ($? != 0) {
diff --git a/git-sh-setup.sh b/git-sh-setup.sh
index c41c2f7..860b164 100755
--- a/git-sh-setup.sh
+++ b/git-sh-setup.sh
@@ -104,14 +104,14 @@ git_editor() {
 	case "$GIT_EDITOR,$TERM" in
 	,dumb)
 		echo >&2 "No editor specified in GIT_EDITOR, core.editor, VISUAL,"
-		echo >&2 "or EDITOR. Tried to fall back to vi but terminal is dumb."
+		echo >&2 "or EDITOR. Tried to fall back to editor but terminal is dumb."
 		echo >&2 "Please set one of these variables to an appropriate"
 		echo >&2 "editor or run $0 with options that will not cause an"
 		echo >&2 "editor to be invoked (e.g., -m or -F for git-commit)."
 		exit 1
 		;;
 	esac
-	eval "${GIT_EDITOR:=vi}" '"$@"'
+	eval "${GIT_EDITOR:=editor}" '"$@"'
 }
 
 is_bare_repository () {
diff --git a/git-svn.perl b/git-svn.perl
index eb4b75a..bd7ec25 100755
--- a/git-svn.perl
+++ b/git-svn.perl
@@ -1321,7 +1321,7 @@ sub get_commit_entry {
 	close $log_fh or croak $!;
 
 	if ($_edit || ($type eq 'tree')) {
-		my $editor = $ENV{VISUAL} || $ENV{EDITOR} || 'vi';
+		my $editor = $ENV{VISUAL} || $ENV{EDITOR} || 'editor';
 		# TODO: strip out spaces, comments, like git-commit.sh
 		system($editor, $commit_editmsg);
 	}
diff --git a/t/t7005-editor.sh b/t/t7005-editor.sh
index b647957..97382ec 100755
--- a/t/t7005-editor.sh
+++ b/t/t7005-editor.sh
@@ -4,7 +4,7 @@ test_description='GIT_EDITOR, core.editor, and stuff'
 
 . ./test-lib.sh
 
-for i in GIT_EDITOR core_editor EDITOR VISUAL vi
+for i in GIT_EDITOR core_editor EDITOR VISUAL editor
 do
 	cat >e-$i.sh <<-EOF
 	#!$SHELL_PATH
@@ -12,15 +12,15 @@ do
 	EOF
 	chmod +x e-$i.sh
 done
-unset vi
-mv e-vi.sh vi
+unset editor
+mv e-editor.sh editor
 unset EDITOR VISUAL GIT_EDITOR
 
 test_expect_success setup '
 
 	msg="Hand edited" &&
 	echo "$msg" >expect &&
-	git add vi &&
+	git add editor &&
 	test_tick &&
 	git commit -m "$msg" &&
 	git show -s --pretty=oneline |
@@ -31,7 +31,7 @@ test_expect_success setup '
 
 TERM=dumb
 export TERM
-test_expect_success 'dumb should error out when falling back on vi' '
+test_expect_success 'dumb should error out when falling back on editor' '
 
 	if git commit --amend
 	then
@@ -44,7 +44,7 @@ test_expect_success 'dumb should error out when falling back on vi' '
 
 TERM=vt100
 export TERM
-for i in vi EDITOR VISUAL core_editor GIT_EDITOR
+for i in editor EDITOR VISUAL core_editor GIT_EDITOR
 do
 	echo "Edited by $i" >expect
 	unset EDITOR VISUAL GIT_EDITOR
@@ -68,7 +68,7 @@ done
 
 unset EDITOR VISUAL GIT_EDITOR
 git config --unset-all core.editor
-for i in vi EDITOR VISUAL core_editor GIT_EDITOR
+for i in editor EDITOR VISUAL core_editor GIT_EDITOR
 do
 	echo "Edited by $i" >expect
 	case "$i" in
-- 
1.6.5.2

